app/src/main/java/org/mozilla/fenix/components/history/PagedHistoryProvider.kt

```diff
@@ -133,6 +135,14 @@ class DefaultPagedHistoryProvider(
                                 items = items.map { it.toHistoryDBMetadata() }
                             )
                         }
+                        .filter {
```

Comment:
To land this we also need to apply this logic to history search groups under "recently visited." Otherwise the logic is inconsistent and we'd have groups in recently visited with a single item, but once the user hits "show all" the groups wouldn't exist in the history view.

Reply:
(Have groups and sizes consistent was important feedback form user testing.)

Reply:
Yes, I have asked about that in the Jira ticket. This is just a draft, while waiting for changes from https://github.com/mozilla-mobile/fenix/pull/23257 to land